### PR TITLE
save and get body and header from file system

### PR DESCRIPTION
**Pull request by danieltigse**
Show email source

### CODE REVIEW

File: iOS-Email-Client/AsyncTasks/CreateCustomJSONFileAsyncTask.swift [hunk 1]
```diff
@@ -33,7 +38,10 @@ class CreateCustomJSONFileAsyncTask {
         results.labels.forEach {handleRow($0.toDictionary())}
         results.emails.enumerated().forEach {
             emails[$1.key] = $0 + 1
-            let dictionary = $1.toDictionary(id: $0 + 1)
+            let dictionary = $1.toDictionary(
+                id: $0 + 1,
+                emailBody: FileUtils.getBodyFromFile(account: myAccount, metadataKey: "\($1.key)"),
```
Comment (Hirobreak) [inline iOS-Email-Client/AsyncTasks/CreateCustomJSONFileAsyncTask.swift]: Can't Access Realm Object From Instantiated in a Different Thread

File: iOS-Email-Client/AsyncTasks/CreateCustomJSONFileAsyncTask.swift [hunk 2]
```diff
@@ -33,7 +38,10 @@ class CreateCustomJSONFileAsyncTask {
...
+                headers: FileUtils.getHeaderFromFile(account: myAccount, metadataKey: "\($1.key)"))
```
Comment (Hirobreak) [inline iOS-Email-Client/AsyncTasks/CreateCustomJSONFileAsyncTask.swift]: Same as above

File: iOS-Email-Client/Libs/FileUtils.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            return
+        }
+        if let fileHandle = try? FileHandle(forUpdating: fileUrl) {
+            fileHandle.seekToEndOfFile()
+            fileHandle.write(data)
+            fileHandle.closeFile()
+        }
+    }
+    
+    static func saveEmailHeadersToFile(account: Account, metadataKey: String, headers: String){
```
Comment (Hirobreak) [inline iOS-Email-Client/Libs/FileUtils.swift]: `saveEmailBodyToFile ` and `saveEmailHeadersToFile ` have the exact same code except for ONE line, abstract this line for the rest of the code and reuse it

File: iOS-Email-Client/Model/DB/Email.swift [hunk 4]
```diff
@@ -170,15 +163,15 @@ class Email: Object {
...
+    func toDictionary(id: Int, emailBody: String, headers: String) -> [String: Any] {
         let dateString = DateUtils().date(toServerString: date)!
         var object = [
             "id": id,
             "messageId": messageId,
             "threadId": threadId,
             "unread": unread,
             "secure": secure,
-            "content": content,
+            "content": emailBody,
```
Comment (Hirobreak) [inline iOS-Email-Client/Model/DB/Email.swift]: check for content if emailBody is empty

File: iOS-Email-Client/Shared/NewEmailHandler.swift [hunk 5]
```diff
@@ -66,12 +66,14 @@ class NewEmailHandler {
...
-                case let .SuccessString(body) = responseData,
+                case let .SuccessDictionary(data) = responseData,
+                let body = data["body"] as? String,
+                let headers = data["headers"] as? String,
                 let username = ContactUtils.getUsernameFromEmailFormat(event.from) else {
                     completion(Result(success: false))
                     return
             }
-            guard let content = unsent ? "" : self.handleBodyByMessageType(event.messageType, body: body, account: myAccount, recipientId: username, senderDeviceId: event.senderDeviceId, isExternal: event.isExternal) else {
+            guard let content = unsent ? "" : self.handleBodyByMessageType(event.messageType, body: body, account: myAccount, recipientId: username, senderDeviceId: event.senderDeviceId, isExternal: event.isExternal), let contentHeader = self.handleheaderByMessageType(event.messageType, header: headers, account: myAccount, recipientId: username, senderDeviceId: event.senderDeviceId, isExternal: event.isExternal) else {
```
Comment (Hirobreak) [inline iOS-Email-Client/Shared/NewEmailHandler.swift]: add line enter, can't see

File: iOS-Email-Client/Shared/NewEmailHandler.swift [hunk 6]
```diff
@@ -165,6 +170,20 @@ class NewEmailHandler {
         return trueBody
     }
     
+    func handleheaderByMessageType(_ messageType: MessageType, header: String, account: Account, recipientId: String, senderDeviceId: Int32?, isExternal: Bool) -> String? {
```
Comment (Hirobreak) [inline iOS-Email-Client/Shared/NewEmailHandler.swift]: handle`H`eader....

File: iOS-Email-Client/Controllers/EmailDetailViewController.swift [hunk 7]
```diff
@@ -197,7 +197,7 @@ extension EmailDetailViewController: UITableViewDelegate, UITableViewDataSource{
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let email = emailData.emails[indexPath.row]
         let cell = reuseOrCreateCell(identifier: "emailDetail\(email.key)") as! EmailTableViewCell
-        cell.setContent(email, state: emailData.getState(email.key), myEmail: emailData.accountEmail)
+        cell.setContent(email, emailBody: self.emailData.bodies[email.key] ?? "", state: emailData.getState(email.key), myEmail: emailData.accountEmail)
```
Comment (Hirobreak) [inline iOS-Email-Client/Controllers/EmailDetailViewController.swift]: what about email.content if emailBody is empty?

File: iOS-Email-Client/Controllers/InboxViewController.swift [hunk 8]
```diff
@@ -1156,7 +1159,7 @@ extension InboxViewController: InboxTableViewCellDelegate, UITableViewDelegate {
         composerData.initToContacts = Array(draft.getContacts(type: .to))
         composerData.initCcContacts = Array(draft.getContacts(type: .cc))
         composerData.initSubject = draft.subject
-        composerData.initContent = draft.content
+        composerData.initContent = FileUtils.getBodyFromFile(account: myAccount, metadataKey: "\(draft.key)")
```
Comment (Hirobreak) [inline iOS-Email-Client/Controllers/InboxViewController.swift]: what about old drafts?

File: iOS-Email-Client/Libs/FileUtils.swift (hunk 3 above)
Comment (danieltigse) [reply iOS-Email-Client/Libs/FileUtils.swift]: TWO lines actually. but OK!

File: iOS-Email-Client/Shared/NewEmailHandler.swift (hunk 5 above)
Comment (danieltigse) [reply iOS-Email-Client/Shared/NewEmailHandler.swift]: I knew you will say that

**Pull request merged**